src/common/mem.c

```diff
@@ -97,10 +303,18 @@ pmem_tx_free(struct umem_instance *umm, umem_off_t umoff)
 }
 
 static umem_off_t
-pmem_tx_alloc(struct umem_instance *umm, size_t size, uint64_t flags,
-	      unsigned int type_num)
+pmem_tx_alloc(struct umem_instance *umm, size_t size, int slab_id,
+	      uint64_t flags, unsigned int type_num)
 {
-	return umem_id2off(umm, pmemobj_tx_xalloc(size, type_num, flags));
+	uint64_t pflags = 0;
+
+	if (flags & UMEM_FLAG_ZERO)
+		pflags |= POBJ_FLAG_ZERO;
```

Comment:
I worry a little about any performance impact this may have.   One possible alternative would be to define these UMEM flags to have same value as the POBJ ones and use 
D_CASSERT(UMEM_FLAG_ZERO == POBJ_FLAG_ZERO);

somewhere in this file.  That's a static, compile time assertion.

Then again, maybe you don't want to do that since these flags could also be used for your bdev implementation.

Reply:
@jolivier23 For IO size of 512 bytes, there is a small increase (300+ns) in the write latency with vos_perf.  I see an improvement with read latency with the fix.

The numbers captured from my test system for 10 runs is shown below:
Without change:
wr lat 10.607us, rd lat 1.888us
wr lat 10.699us, rd lat 1.897us
wr lat 10.513us, rd lat 1.866us
wr lat 10.669us, rd lat 1.958us
wr lat 10.858us, rd lat 1.864us
wr lat 10.889us, rd lat 1.937us
wr lat 10.693us, rd lat 1.896us
wr lat 10.575us, rd lat 1.800us
wr lat 10.667us, rd lat 1.826us
wr lat 10.565us, rd lat 1.807us

With change:
wr lat 10.942us, rd lat 1.551us
wr lat 10.981us, rd lat 1.532us
wr lat 11.134us, rd lat 1.530us
wr lat 11.127us, rd lat 1.469us
wr lat 11.064us, rd lat 1.534us
wr lat 10.957us, rd lat 1.528us
wr lat 11.172us, rd lat 1.489us
wr lat 10.954us, rd lat 1.548us
wr lat 11.068us, rd lat 1.547us
wr lat 10.906us, rd lat 1.505us

Yes, the UMEM flags are currently same as that of POBJ flags. I tried the D_CASSERT() recommendation and it did not make a difference in the vos_perf numbers. If there are no other comments that require a rework, is it ok I make this change in new/future PR?